docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,697 @@
+    * **type**: provides information about the way IP was provided
+  * **addresses**: section for generated domain
+    * **hostname**: generated domain 
+    * **status**: indicate if an address is available
+    * **origin**: section providing information what generated the vip
+      * **kind**: points to kind that generated domain
+      * **name**: name of the `HostnameGenerator`
+    * **reason**: holds error messages if there are any 
+
+#### Default CA certificate
+
+By default, when TLS is enabled and CA certificate is not set, we are going to use system bundled one. We can achive that by scanning in `kuma-dp` [default paths](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#example-configuration) for each environment and later send that information in the bootstrap request to the control-plane for a configuration generation.
+
+#### Policy matching
+
+`MeshExternalServices` should work the same with policies as `MeshService`. We are going to introduce a new `kind: MeshExternalService`, which allows targeting them with policies.
+
+#### Envoy resources naming
+
+Currently, we name clusters after their respective policies. Since the model includes only one match we can stick to the same pattern but prefix it with `mes_` which stands for `MeshExternalService`.
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
+  managed:
+    match:
+      type: Domain # Kuma will generate a domain
+      value: httpbin.com
+      port: 80
+      protocol: http
+    extension:
+      type: Lambda 
+      config: # type JSON
+        arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+    endpoints:
+      - address: 1.1.1.1
+        port: 12345
+```
+
+Cluster name: `mes_{policyName}`, so for the policy name above, it would look like: `mes_httpbin`.
+
+Listener name: We are going to create listeners only for `KumaHostname` and `IP`. In this case, the naming remains the same as now: `outbound:{address}:{port}`.
+
+For `CIDR` and `Domain`, we will create filter chain matches to exclude them from the traffic. For TLS and TCP traffic, we will name them `mes_{policyName}` because each listener points to the cluster. For `HTTP` traffic, we will name them `mes_http_{port}` because we can distinguish the traffic on routes based on host header.
+
+#### Extensability
+
+Provided model allows creating separate plugins which can integrate with different cloud providers or components. The user can create it's own custom configuration and later in the code create XDS configuration based on it. 
+
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: lambda-example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
+  managed:
+    match:
+      type: KumaHostname # Kuma will generate a domain
```

Comment:
Before it was `Domain`. What's the difference?

Reply:
Still `Domain`, `IP`, `CIDR` are available, now instead of `InternalVIP` we renamed it `KumaHostname`

Reply:
@lukidzi are you sure they make sense? we have them in https://github.com/kumahq/kuma/pull/10229/files#r1607739352